### PR TITLE
Added extension nft meta data pointer example

### PR DESCRIPTION
**Pull request by Woody4618**
Added an example on how to create NFTs with the token meta data pointer extension and additional meta data with nextjs app, unity client and anchor program.

### CODE REVIEW

**Review by brimigs — COMMENTED** [review]
just some feedback on general rust 👍

File: tokens/token-2022/nft-meta-data-pointer/anchor/program/programs/extension_nft/src/instructions/chop_tree.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        mut,
+        seeds = [b"player".as_ref(), player.authority.key().as_ref()],
+        bump,
+    )]
+    pub player: Account<'info, PlayerData>,
+
+    // There can be multiple levels the seed for the level is passed in the instruction
+    // First player starting a new level will pay for the account in the current setup
+    #[account(
+        init_if_needed,
```
[the remark this answers is not in the document]
Comment (Woody4618) [reply tokens/token-2022/nft-meta-data-pointer/anchor/program/programs/extension_nft/src/instructions/chop_tree.rs:76]: Not quite sure, what you mean here. For the game_data init_if_needed implies mutable and the signer is marked with 
´´´
    #[account(mut)]
´´´

File: tokens/token-2022/nft-meta-data-pointer/anchor/program/programs/extension_nft/src/instructions/chop_tree.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
+pub use crate::errors::GameErrorCode;
+pub use crate::state::game_data::GameData;
+use crate::{state::player_data::PlayerData, NftAuthority};
+use anchor_lang::prelude::*;
+use anchor_spl::token_interface::{spl_token_2022::state::Mint, Token2022};
+use session_keys::{Session, SessionToken};
+use solana_program::program::invoke_signed;
+
+pub fn chop_tree(mut ctx: Context<ChopTree>, counter: u16, amount: u64) -> Result<()> {
+    let account: &mut &mut ChopTree<'_> = &mut ctx.accounts;
```
[the remark this answers is not in the document]
Comment (Woody4618) [reply tokens/token-2022/nft-meta-data-pointer/anchor/program/programs/extension_nft/src/instructions/chop_tree.rs:10]: Oh yes true, thanks!

File: tokens/token-2022/nft-meta-data-pointer/anchor/program/programs/extension_nft/src/state/game_data.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+use crate::constants::MAX_WOOD_PER_TREE;
+
+#[account]
+pub struct GameData {
+    pub total_wood_collected: u64,
+}
+
+impl GameData {
+    pub fn on_tree_chopped(&mut self, amount_chopped: u64) -> Result<()> {
+        match self.total_wood_collected.checked_add(amount_chopped) {
```
[the remark this answers is not in the document]
Comment (Woody4618) [reply tokens/token-2022/nft-meta-data-pointer/anchor/program/programs/extension_nft/src/state/game_data.rs:12]: Isnt the "match" doing exactly that check? If none its an overflow if some it worked. Or are there also other cases then it returns none?

**Review by brimigs — APPROVED** [review]

**Pull request merged**